src/Altinn.Platform/Altinn.Platform.Events/Events/Repository/PostgresRepository.cs

```diff
@@ -227,5 +229,89 @@ public async Task<Subscription> GetSubscription(int id)
+                    {
+                        Subscription subscription = new Subscription();
+                        subscription.Id = Convert.ToInt32(reader["id"].ToString());
+                        subscription.SourceFilter = new Uri(reader["sourcefilter"].ToString());
+                        subscription.SubjectFilter = reader["subjectfilter"].ToString();
+                        subscription.TypeFilter = reader["typefilter"].ToString();
+                        subscription.Consumer = reader["consumer"].ToString();
+                        subscription.EndPoint = new Uri(reader["endpointurl"].ToString());
+                        subscription.CreatedBy = reader["createdby"].ToString();
+                        subscription.Created = DateTime.Parse(reader["time"].ToString());
+                        searchResult.Add(subscription);
+                    }
+                }
+
+                return searchResult;
+            }
+            catch (Exception e)
+            {
+                _logger.LogError("PostgresRepository // GetSubscriptionsExcludeOrg // Exception", e);
+                throw;
+            }
+            finally
+            {
+                await _conn.CloseAsync();
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<List<Subscription>> GetSubscriptionsByConsumer(string consumer)
+        {
+            List<Subscription> searchResult = new List<Subscription>();
+            try
+            {
+                await _conn.OpenAsync();
+
+                NpgsqlCommand pgcom = new NpgsqlCommand(getSubscriptionByConsumerSql, _conn);
+                pgcom.Parameters.AddWithValue("_consumer", NpgsqlDbType.Varchar, consumer);
+
+                using (NpgsqlDataReader reader = pgcom.ExecuteReader())
+                {
+                    while (reader.Read())
+                    {
+                        Subscription subscription = new Subscription();
+                        subscription.Id = Convert.ToInt32(reader["id"].ToString());
+                        subscription.SourceFilter = new Uri(reader["sourcefilter"].ToString());
+                        subscription.SubjectFilter = reader["subjectfilter"].ToString();
+                        subscription.TypeFilter = reader["typefilter"].ToString();
+                        subscription.Consumer = reader["consumer"].ToString();
+                        subscription.EndPoint = new Uri(reader["endpointurl"].ToString());
+                        subscription.CreatedBy = reader["createdby"].ToString();
+                        subscription.Created = DateTime.Parse(reader["time"].ToString());
+                        searchResult.Add(subscription);
+                    }
+                }
+
+                return searchResult;
+            }
+            catch (Exception e)
+            {
+                _logger.LogError("PostgresRepository // GetSubscriptionByConsumer // Exception", e);
+                throw;
+            }
```

Comment:
This catch might be unnecessary. I can't see any other try catch that would handle it. This means the exception will pass through all the way up to the Web API middleware and be handled by app wide exception logging.

There is always the chance that someone suddenly adds a catch and "ignores" the exception higher up. Safety net for future changes? (I'm personally a little undecided on the matter.)